htseq2multx.pl

```diff
@@ -0,0 +1,884 @@
+  --galaxy              Produce "Galaxy safe" filenames by removing
+                        underscores (default: False)
+  --nosanitize          Do not produce "safe" filenames by replacing unusual
+                        characters in the supplied prefix and sample IDs with
+                        underscores. (default: False)
+  -v, --verbose         verbose output
+  --gzipout             Output files in compressed gzip format (default is
+                        uncompressed)
+  --split_all           Split all input files, including index read files (by
+                        default, index read files are not split unless all
+                        read files are index files)
+
+Input format:
+  --format FORMAT       Specify format for sequence files (fasta or fastq)
+  --gzipin              Assume input files are in gzip format, despite file
+                        extension (default is auto based on input file
+                        extension)
+
+Sequence Inputs:
+  FILE                  A series of 1 or more [optionally zipped] fastq files.
+END_LONG
+
+    print($short_msg);
+    print($long_msg) unless($short_only);
+  }
+
+sub help2
+  {
+    my $msg = << 'END_HELP2';
+
+The following extra and modified options apply specifically to the htseq2multx.pl wrapper (as opposed to barcode_splitter).
+
+optional arguments:
+  -h, --help            Supply twice to show this message and exit.
+  --version             Supply once to get the barcode_splitter version that
+                        this wrapper emulates.  Supply twice to get a more
+                        verbose version of the barcode_splitter version that
+                        this wrapper emulates and the version of this wrapper.
+
+configuration arguments:
+  --fastq-multx PATH    Path to the fastq-multx installation.  Default is to use
+                        the PATH in the user's environment.
+  --debug               Supply this flag to output the fastq-multx command on
+                        standard error and not delete the temporary barcode
+                        file.
+END_HELP2
+
+    print($msg);
+  }
+
+sub getFastqMultxCommand
+  {
+    my $command =
+      "$fastq_multx -x -d 1 -B '$bcfilefqmx' -m $mismatches -M $mismatches";
+
+    $command .= ($barcodes_at_end ? ' -e' : ' -b');
+    $command .= ($check_deflines  ? " -v '$deflinesep'" : '');
+
+    #Add the input files and output file templates
+    my $templates_str = '';
+    foreach my $fqf_index (sort {indexesFirst($a,$b)}
+                           0..$#{$fastq_files})
+      {
+        $command       .= " '$fastq_files->[$fqf_index]'";
+        $templates_str .= " -o '" . getOutfileTemplate($fqf_index) . "'";
+      }
+    $command .= $templates_str;
+
+    return($command);
+  }
+
+#Runs the command and prints stdout/stderr.  Exits as barcode_splitter would.
+sub runFastqMultx
+  {
+    my $command = $_[0];
+
+    my $producer = new IO::Pipe::Producer();
```

Comment:
Don't use indirect syntax: http://modernperlbooks.com/mt/2009/08/the-problems-with-indirect-object-notation.html

```suggestion
    my $producer = IO::Pipe::Producer()->new;
```

Reply:
Interesting.  I should probably update my module on CPAN too.

- [ ] Change `new ...Producer` to `Producer->new`
- [ ] Change the CPAN module documentation to reflect the above

I had noted some other doc improvements that I need to make to that module.  Incidentally, that's the first module I ever put on CPAN.